rero_ils/modules/monitoring.py

```diff
@@ -0,0 +1,309 @@
+    return jsonify({'data': Monitoring.info()})
+
+
+@api_blueprint.route('/check_es_db_counts')
+def check_es_db_counts():
+    """Displays health status for ES and DB counts."""
+    result = {'data': {'status': 'green'}}
+    check = Monitoring.check()
+    if check:
+        details = []
+        links = {'about': url_for(
+            'api_monitoring.check_es_db_counts', _external=True)}
+        for info, count in check.items():
+            msg = 'There are {count} items from {info} missing in ES.'.format(
+                count=count,
+                info=info
+            )
+            details.append(msg)
+            links[info] = url_for(
+                'api_monitoring.missing_pids',
+                doc_type=info,
+                _external=True
+            )
+        result = {
+            'data': {'status': 'red'},
+            'error': {
+                'id': 'DB_ES_COUNTER_MISSMATCH',
+                'links': links,
+                'code': 'DB_ES_COUNTER_MISSMATCH',
+                'title': "DB items counts don't match ES items count.",
+                'details': details
+            }
+        }
+    return jsonify(result)
+
+
+def check_authentication(func):
+    """Decorator to check authentication for items HTTP API."""
+    @wraps(func)
+    def decorated_view(*args, **kwargs):
+        if not current_user.is_authenticated:
+            return jsonify({'status': 'error: Unauthorized'}), 401
+        if not admin_permission.require().can():
+            return jsonify({'status': 'error: Forbidden'}), 403
+        return func(*args, **kwargs)
+
+    return decorated_view
+
+
+@api_blueprint.route('/missing_pids/<doc_type>')
+@check_authentication
+def missing_pids(doc_type):
+    """Displays missing pids for doc type.
+
+    :param doc_type: Document type to display.
+    """
+    try:
+        api_url = url_for(
+            'invenio_records_rest.{doc_type}_list'.format(doc_type=doc_type),
+            _external=True
+        )
+    except:
+        api_url = None
+    mon = Monitoring.missing(doc_type)
+    data = {'DB': [], 'ES': [], 'ES double': {}}
+    for pid in mon.get('DB'):
+        if api_url:
+            data['DB'].append(
+                '{api_url}?q=pid:{pid}'.format(api_url=api_url, pid=pid))
+        else:
+            data['DB'].append(pid)
+    for pid in mon.get('ES'):
+        if api_url:
+            data['ES'].append(
+                '{api_url}{pid}'.format(api_url=api_url, pid=pid))
+        else:
+            data['ES'].append(pid)
+    for pid in mon.get('ES double'):
+        if api_url:
+            url = '{api_url}?q=pid:{pid}'.format(api_url=api_url, pid=pid)
+            count = data['ES double'].setdefault(url, 0)
+            data['ES double'][url] = count + 1
+        else:
+            count = data['ES double'].setdefault(pid, 0)
+            data['ES double'][pid] = count + 1
+    return jsonify({'data': data})
+
+
+class Monitoring(object):
+    """Monitoring class."""
```

Comment:
can you put more details in the comment. The main idea here is to check the consistency between the database and the search index. We need to check that all documents presents in the database are also present in the search index and vice versa.